tests/page/selectors-frame.spec.ts

```diff
@@ -145,3 +198,60 @@ it('should click in lazy iframe', async ({ page, server }) => {
   ]);
   expect(text).toBe('Hello iframe');
 });
+
+it('waitFor should survive frame reattach', async ({ page, server }) => {
+  await routeIframe(page);
+  await page.goto(server.EMPTY_PAGE);
+  const button = page.locator('iframe >> content-frame=true >> button:has-text("Hello nested iframe")');
+  const promise = button.waitFor();
+  await page.locator('iframe').evaluate(e => e.remove());
+  await page.evaluate(() => {
+    const iframe = document.createElement('iframe');
+    iframe.src = 'iframe-2.html';
+    document.body.appendChild(iframe);
+  });
+  await promise;
+});
+
+it('click should survive frame reattach', async ({ page, server }) => {
+  await routeIframe(page);
+  await page.goto(server.EMPTY_PAGE);
+  const button = page.locator('iframe >> content-frame=true >> button:has-text("Hello nested iframe")');
+  const promise = button.click();
+  await page.locator('iframe').evaluate(e => e.remove());
+  await page.evaluate(() => {
+    const iframe = document.createElement('iframe');
+    iframe.src = 'iframe-2.html';
+    document.body.appendChild(iframe);
+  });
+  await promise;
+});
+
+it('click should survive iframe navigation', async ({ page, server }) => {
+  await routeIframe(page);
+  await page.goto(server.EMPTY_PAGE);
+  const button = page.locator('iframe >> content-frame=true >> button:has-text("Hello nested iframe")');
+  const promise = button.click();
+  page.locator('iframe').evaluate(e => (e as HTMLIFrameElement).src = 'iframe-2.html');
+  await promise;
+});
+
+it('click should survive navigation', async ({ page, server }) => {
+  await routeIframe(page);
+  await page.goto(server.PREFIX + '/iframe.html');
+  const promise = page.click('button:has-text("Hello nested iframe")');
+  await page.waitForTimeout(100);
+  await page.goto(server.PREFIX + '/iframe-2.html');
+  await promise;
+});
+
+it('click fail if waiting for selector on element handle', async ({ page, server }) => {
+  it.fixme();
```

Comment:
How does this test fail? Also, title has "click" but there is no click in the test.

Reply:
renamed the test